torch/custom_class.h

```diff
@@ -221,6 +221,82 @@ class class_ {
     return *this;
   }
 
+  // This is a temporary hack to change the format of ConvPackedParams without
+  // breaking BC. We modify the default pickle protocol by defining a boxed
+  // version of __set_state__, which knows how to parse historical formats.
+  // TODO(before land): create an issue to delete this in a month and link here.
+  template <typename GetStateFn, typename SetStateFn>
+  class_& def_pickle_unboxed_DO_NOT_USE(
+      GetStateFn&& get_state, SetStateFn&& set_state) {
```

Comment:
you can simplify the code quite a bit by explicitly spelling out SetStateFn as `std::function<intrusive_ptr<CurClass>(IValue)>` and avoid templates. It's way more explicit